File: command-pattern-master/command-pattern-master/c++/sdk/robot.cpp
/*
 * robot.cpp
 *
 *  Created on: Nov 15, 2017
 *      Author: ramiz
 */

#include <iostream>
#include <math.h>
#include "robot.h"

Robot::Robot() {
  this->current_angle_ = 0;
  this->current_robot_position_ = 0;
}

Robot::~Robot() {

}

void Robot::Move(double distance) {
  current_robot_position_ += distance;
  printf("Moved robot to position %f\n", current_robot_position_);
}

void Robot::Turn(double angle) {
  current_angle_ += angle;
  printf("Turned robot to angle %f\n", current_angle_);
}

void Robot::Beep() {
  printf("Robot beeping\n");
}

File: command-pattern-master/command-pattern-master/java/src/sdk/Robot.java
package sdk;

/**
 * Created by ramiz on 11/15/17.
 */
public class Robot implements IRobot {
    private double currentPosition = 0.0;
    private double currentAngle = 0.0;

    @Override
    public void moveRobot(double step) {
        currentPosition += step;
        System.out.println("Moved robot to position " + currentPosition);
    }

    @Override
    public void turnRobot(double angleStep) {
        currentAngle += angleStep;
        currentAngle %= 2 * Math.PI;
        System.out.println("Turned robot to angle " + currentAngle);
    }

    @Override
    public void beepRobot() {
        System.out.println("Robot beeping");
    }
}
